pkgs/misc/emulators/retroarch/fix-paths.patch

```diff
@@ -0,0 +1,19 @@
+diff --git a/configuration.c b/configuration.c
+index e6a3841324..5c6027dcea 100644
+--- a/configuration.c
++++ b/configuration.c
+@@ -2919,6 +2919,14 @@ void config_load(void *data)
+ #ifdef HAVE_CONFIGFILE
+    config_parse_file(global);
+ #endif
++   // Ignore config paths and use the values from /nix/store instead
++   settings_t *settings = config_st;
++   fill_pathname_expand_special(settings->paths.directory_libretro,
++         "@libretro_directory@",
++         sizeof(settings->paths.directory_libretro));
++   fill_pathname_expand_special(settings->paths.path_libretro_info,
++         "@libretro_info_path@",
++         sizeof(settings->paths.path_libretro_info));
```

Comment:
Another option would be to patch `retroarch.cfg` like I did with the `menu_show_core_updater`. However this would create an issue that every time someone bumped this package, they would need to delete their `retroarch.cfg` file, or at least remove the `libretro_diretory` and `libretro_info_path` options.

Reply:
Yet another option would be to add a module that creates a `/etc/retroarch.cfg` file with the correct path. This is also a good solution, but wouldn't work on non-NixOS.

Reply:
This solution is not perfect though, for example, the wrong values are still written on the `retroarch.cfg` file (however they're ignored). This may be confusing for some people.